src/endpoint/s3/ops/s3_post_object_restore.js

```diff
@@ -1,32 +1,44 @@
 /* Copyright (C) 2023 NooBaa */
 'use strict';
 
+const { S3Error } = require('../s3_errors');
 const s3_utils = require('../s3_utils');
 
 /**
  * https://docs.aws.amazon.com/AmazonS3/latest/API/API_RestoreObject.html
  */
 async function post_object_restore(req, res) {
     const encryption = s3_utils.parse_encryption(req);
-    const days = req.body?.RestoreRequest?.Days ? parseInt(req.body.RestoreRequest.Days?.[0], 10) : undefined;
+
+    if (!req.body?.RestoreRequest?.Days?.[0]) {
+        throw new S3Error(S3Error.MalformedXML);
```

Comment:
please add a dbg.warn with more info, otherwise it's really hard to figure out.
An even better thing to do would be to add a description to the error message that explains what caused the validation failure. But at least logging is a must.